### ISSUE TITLE
[Bug] Shrinking the past-epoch retention keeps the wrong epochs

### ISSUE THREAD

**Issue by mtrossbach** [openmls/openmls#2196]
**Describe the bug**

`MessageSecretsStore::resize` rotates the store before truncating it, so that `truncate` drops the oldest entries. It rotates by the difference between the old and the new limit, which equals the number of entries to drop only while the store is full (`openmls/src/group/mls_group/past_secrets.rs:85`):

```rust
let old_size = self.max_epochs;
self.max_epochs = max_past_epochs;
if old_size > max_past_epochs {
    let num_epochs_out = old_size - max_past_epochs;
    self.past_epoch_trees
        .rotate_left(num_epochs_out.min(self.past_epoch_trees.len()));
    self.past_epoch_trees.truncate(max_past_epochs);
}
```

Below that the rotation overshoots. Working through the algorithm gives three failure modes, all of which leave the store out of order: a window that is not contiguous, the oldest `n` where `min(len)` clamps the overshoot to a full turn, and, where nothing needs dropping, the right epochs reversed.

**To Reproduce**

Through `MlsGroup::set_past_epoch_deletion_policy`, or `set_configuration` with a changed policy. The store starts empty on creation and on every join, so any shrink before it has filled is affected; under `KeepAll` its limit is `isize::MAX`, so it never fills.

1. `MaxEpochs(6)`, advance 5 epochs, shrink to 3.
2. `KeepAll`, advance 30 epochs, shrink to 3.
3. `MaxEpochs(3)`, advance 2 epochs, shrink to 2, then commit once.

**Expected behavior**

The `n` most recent past epochs, in order.

**Actual behavior**

| case | retained before | retained after | expected |
| --- | --- | --- | --- |
| `MaxEpochs(6)`, 5 epochs, shrink to 3 | `0 … 4` | `0, 3, 4` | `2, 3, 4` |
| `KeepAll`, 30 epochs, shrink to 3 | `0 … 29` | `0, 1, 2` | `27, 28, 29` |
| `MaxEpochs(10)`, store full, shrink to 2 | `2 … 11` | `10, 11` | `10, 11` ✓ |

A shrink that drops nothing is not harmless either. With `MaxEpochs(3)` and 2 epochs, shrinking to 2 removes neither, but the store is left reversed, so the next commit evicts epoch 0 instead of epoch 1 and leaves `0, 2` where `1, 2` is correct.

The store is written through to the storage provider, so the loss is permanent. Messages from the missing epochs fail with `SecretTreeError::TooDistantInThePast`, the same error #2180 reports for a different cause.

**Screenshots or debug log**

Nothing is logged; the truncation is silent. Measured by listing the epochs for which `message_secrets_and_leaves` returns `Ok`.

**Platform (please complete the following information):**

- OS & OS version: macOS 15 (Darwin 25.5.0); not platform-specific
- Architecture: aarch64
- Version: `main` @ 00ef45ba, default features
- Ciphersuite: `MLS_128_DHKEMX25519_AES128GCM_SHA256_Ed25519`, `openmls_rust_crypto`

**Additional context**

This is a regression rather than a decision. `resize` began as `while len > max { pop_front() }`. 28a0a45c replaced that with the rotation and substituted `old_size - max_past_epochs` for the excess, and #1972 added the `.min(len)` clamp when it made `resize` reachable at runtime, where the overshoot would otherwise panic.

`add_past_epoch_tree` is unaffected, its excess is always exactly one, and `delete_past_epoch_secrets` takes its from `len()`. Doing the same in `resize` fixes every case:

```rust
let excess = self.past_epoch_trees.len().saturating_sub(max_past_epochs);
self.past_epoch_trees.drain(0..excess);
```

`test_update_policy` shrinks only from a full store, which is the case the current code gets right, and asserts on the count. A PR with this fix and tests that assert on the surviving epochs follows.
